### PR TITLE
Statefulcomps

### PR DESCRIPTION
**Pull request by akshay-narkar**
- App component is converted to a stateful component
- Button panel passes a clickHandler to the buttons
- Event listners are triggered lifting states to stateful App component
- Next value is passed to display to display the result

## Changes based on 1st review
- Remove decimal from the constructor
- Refactor App and calculate code

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -3,20 +3,143 @@ import ButtonPanel from './ButtonPanel';
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+      decimal: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
```
Comment (voscarmv) [inline src/components/App.js]: - [x] The project does not require a `decimal` option for the constructor. This only adds unnecessary complexity to the calculation. Please see the example of [my own solution](https://github.com/voscarmv/calculator/blob/development/src/components/App.js#:~:text=extends%20React.Component%20%7B-,constructor(props)%20%7B,%7D,-render()%20%7B) to this project in order to see how you may simplify this part.

File: src/components/App.js [hunk 2]
```diff
@@ -3,20 +3,143 @@ import ButtonPanel from './ButtonPanel';
...
-      <Display result="25" />
-    </>
-  );
-}
+          ));
+        } else {
+          this.setState(() => ({
+            next: calc.next1,
+          }
+
```
Comment (voscarmv) [inline src/components/App.js]: - [x] Please move all the logic from the calculation to a different file, to keep your code organized. Again, you may follow the example from [my solution](https://github.com/voscarmv/calculator/blob/7aa17ac397ff1a3bf53f335d21568d0524d288e4/src/logic/calculate.js#L3).

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work.  💻 👀 ✨

## However, there are some changes required 👀 🧰 🔨 🔧
- Please read my comments on how to simplify your solution to the project, and the proper structure of the app component constructor.

### Submit these changes and request a new code review! 😄 👍

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier)🧑‍💻🗨️🧑‍💻
# Happy Coding! 🧠 ⌨️ 🖱️ 🖥️

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy:  :balloon: 

Hi @akshay-narkar, congratulations on this milestone so far :smile: Having move the logic into the calculate.js file and the reomval of the decimal in the constructor :+1: I hereby approve the project :smile: Kindly go ahead and merge in to the dev branch :fireworks: 

[Optional]
- The operations sign is not displayed in the calculator display, which is making the app a little awkward. It would be great if we fix that

_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**